metricbeat/module/windows/wmi/wmi.go

```diff
@@ -0,0 +1,212 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//go:build windows
+
+package wmi
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/elastic/beats/v7/libbeat/common/cfgwarn"
+	"github.com/elastic/beats/v7/metricbeat/mb"
+	"github.com/elastic/elastic-agent-libs/mapstr"
+
+	wmi "github.com/microsoft/wmi/pkg/wmiinstance"
+)
+
+// init registers the MetricSet with the central registry as soon as the program
+// starts. The New function will be called later to instantiate an instance of
+// the MetricSet for each host is defined in the module's configuration. After the
+// MetricSet has been created then Fetch will begin to be called periodically.
+func init() {
+	mb.Registry.MustAddMetricSet("windows", "wmi", New)
+}
+
+// MetricSet holds any configuration or state information. It must implement
+// the mb.MetricSet interface. And this is best achieved by embedding
+// mb.BaseMetricSet because it implements all of the required mb.MetricSet
+// interface methods except for Fetch.
+type MetricSet struct {
+	mb.BaseMetricSet
+	config Config
+}
+
+const WMIDefaultNamespace = "root\\cimv2"
+
+// New creates a new instance of the MetricSet. New is responsible for unpacking
+// any MetricSet specific configuration options if there are any.
+func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
+	cfgwarn.Beta("The windows wmi metricset is beta.")
+
+	config := NewDefaultConfig()
+	if err := base.Module().UnpackConfig(&config); err != nil {
+		return nil, err
+	}
+
+	err := config.ValidateConnectionParameters()
+	if err != nil {
+		return nil, err
+	}
+
+	err = config.CompileQueries()
+	if err != nil {
+		return nil, err
+	}
+
+	err = config.ApplyDefaultNamespaceToQueries(config.Namespace)
+	if err != nil {
+		return nil, err
+	}
+
+	config.BuildNamespaceQueryIndex()
+
+	if config.WarningThreshold == 0*time.Second {
```

Comment:
nit: 0*time.Second == 0, so just `0` is enough.

Reply:
do we document anywhere that this default to the collection period?

Reply:
just wanted follow up to ensure this gets addressed in the future - we should ensure we document that there is a default timeout period.